### PR TITLE
feat: Entity reference support

### PR DESCRIPTION
**Pull request by eriksvedang**

### CODE REVIEW

File: src/template/csharp/unity/unity-send-system.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ namespace {{.InternalNamespace}}
         {
             Debug.Log($"created messageChannels");
             var netSys = World.GetOrCreateSystem<NetworkSystem>();
+            CoherenceToUnityConverters.networkSystem = netSys;
```
Comment (rpadrela) [inline src/template/csharp/unity/unity-send-system.go]: Do you know the reason for CoherenceToUnityConverters to need to know about the network system?

Comment (eriksvedang) [reply src/template/csharp/unity/unity-send-system.go]: This code is outdated.

Comment (eriksvedang) [reply src/template/csharp/unity/unity-send-system.go]: You must be looking at an old commit, that was a temporary hack to get things off the ground.

**Review by kingstone426 — COMMENTED** [review]
It seems most of the code changes concern passing a reference to the UnityMapper through lots and lots of instantiated classes that used to be static. Could you achieve the same using GetExistingSystem<NetworkSystem>().Mapper only where it is needed?

File: src/template/csharp/unity/unity-send-system.go (hunk 1 above)
Comment (rpadrela) [reply src/template/csharp/unity/unity-send-system.go]: I see that it gets removed in a subsequent commit.

**Comment by eriksvedang** [comment]
> It seems most of the code changes concern passing a reference to the UnityMapper through lots and lots of instantiated classes that used to be static. Could you achieve the same using GetExistingSystem().Mapper only where it is needed?

Maybe, but I wanted the code to support multiple Mappers, possibly. I had it working with an ugly global reference but it felt too hacky, so I made this change which is a lot more maintainable I think.

File: src/generate/create.go [hunk 2]
```diff
@@ -124,6 +127,13 @@ func convertType(enums map[string]*Enum, fieldType string, language ProgrammingL
 		default:
 			return "Piot.Basal.Quaternion", quaternionString, nil
 		}
+	case "Entity":
+		switch language {
+		case Go:
+			return "ecs.Entity", quaternionString, nil
```
Comment (rpadrela) [inline src/generate/create.go:133]: is quaternionString here correct?

Comment (eriksvedang) [reply src/generate/create.go:133]: No, that is wrong! Good catch.

**Review by kingstone426 — COMMENTED** [review]
Where does the magic happen? :) Can't find where the mapper translates IDs back & forth.

**Comment by eriksvedang** [comment]
> Where does the magic happen? :) Can't find where the mapper translates IDs back & forth.

Heh, yeah... :)
See https://github.com/coherence/field-stream-dotnet/pull/12/files 
and https://github.com/coherence/protocol-definition-dotnet/pull/7/files

**Comment by rpadrela** [comment]
@kingstone426 , will leave it to you to approve as I already had a look and couldn't find any issues.

**Review by kingstone426 — APPROVED** [review]

**Pull request merged**